lib/src/configuration.dart

```diff
@@ -29,6 +29,12 @@ class Configuration {
   /// Creates an implicit configuration with the given [values].
   Configuration.implicit(this._values);
 
+  /// Uniquely identifying ID for a configuration lasting across the execution
+  /// context.
+  ///
+  /// Implicit configurations will always have different IDs.
+  int get opaqueId => identityHashCode(this);
```

Comment:
Hash codes aren't guaranteed to be globally unique, although it may not be relevant; see below.

Reply:
Per my understanding the `identityHashCode` returns the `Object.hashCode` regardless of the object overriding it, and it should generate a different one for each instance even if they have the exact same values 🤔

Reply:
It may be the case that `identityHashCode` produces unique integers for each object today, but the contract of a hash code is only good enough to make it work for a hash table, which means that collisions are allowed as long as they're sufficiently unlikely. The documentation for [`Object.hashCode`](https://api.dart.dev/stable/dart-core/Object/hashCode.html) even explicitly says:

> Objects that are not equal are allowed to have the same hash code.

Reply:
I was trying to avoid importing a new dependency since dart doesn't have a way of doing it (flutter does), but I ended up using `xid`